go-controller/pkg/node/node_ip_handler_linux.go

```diff
@@ -25,17 +26,22 @@ type addressManager struct {
 	nodeAnnotator  kube.Annotator
 	mgmtPortConfig *managementPortConfig
 
+	// compare node primary IP change
+	nodePrimaryAddr net.IP
+	gatewayBridge   *bridgeConfiguration
+
 	OnChanged func()
 	sync.Mutex
 }
 
 // initializes a new address manager which will hold all the IPs on a node
-func newAddressManager(nodeName string, k kube.Interface, config *managementPortConfig, watchFactory factory.NodeWatchFactory) *addressManager {
+func newAddressManager(nodeName string, k kube.Interface, config *managementPortConfig, watchFactory factory.NodeWatchFactory, gwBridge *bridgeConfiguration) *addressManager {
```

Comment:
I'm worried this isn't thread safe. The gwBridge *bridgeConfiguration is also used by the openflow manager thread. wdyt?

Reply:
@trozet good point! In current implementation, it may be safe that the `gwBridge.bridgeName` in AddressManager is used for retrieving the primary interface IP addresses and updating node annotation, we didn't modify any fields (e.g. ips) in gwBridge that are used by openflow manager. In case it does need to modify gwBridge in AddressManager in the future, I think we can add a function to update the openflowManager struct data in the nodeIPManager.onChanged callback. Does it sound reasonable or maybe you have other considerations?

Reply:
I'm thinking in the future other threads may also read/write from *bridgeConfiguration. Should we just add a mutex to that struct now to protect it?

Reply:
Added the sync.Mutex in bridgeConfiguration struct.